Extension/src/Utility/Text/scanner.ts

```diff
@@ -0,0 +1,1018 @@
+
+                case CharacterCodes.equals:
+                    return this.isConflictMarker() ?
+                        this.next(Kind.ConflictMarker, mergeConflictMarkerLength) :
+                        this.#chNext === CharacterCodes.equals ?
+                            this.#chNextNext === CharacterCodes.equals ?
+                                this.next(Kind.EqualsEqualsEquals, 3) :
+                                this.next(Kind.EqualsEquals, 2) :
+                            this.#chNext === CharacterCodes.greaterThan ?
+                                this.next(Kind.EqualsArrow, 2) :
+                                this.next(Kind.Equals);
+
+                case CharacterCodes.bar:
+                    return this.isConflictMarker() ?
+                        this.next(Kind.ConflictMarker, mergeConflictMarkerLength) :
+                        this.#chNext === CharacterCodes.bar ?
+                            this.#chNextNext === CharacterCodes.equals ?
+                                this.next(Kind.BarBarEquals, 3) :
+                                this.next(Kind.BarBar, 2) :
+                            this.#chNext === CharacterCodes.equals ?
+                                this.next(Kind.BarEquals, 2) :
+                                this.next(Kind.Bar);
+
+                case CharacterCodes.singleQuote:
+                case CharacterCodes.doubleQuote:
+                    return this.scanString();
+
+                case CharacterCodes.backtick:
+                    return this.#column === 0 && this.#chNext === CharacterCodes.backtick && this.#chNextNext === CharacterCodes.backtick ?
+                        this.scanCodeFence() : this.scanString();
+
+                default:
+                    // FYI:
+                    // Well-known characters that are currently not processed
+                    //   # \
+                    // will need to update the scanner if there is a need to recognize them
+                    return isIdentifierStart(this.#ch) ? this.scanIdentifier() : this.next(Kind.Unknown);
+            }
+        }
+
+        this.text = '';
+        return this.kind = Kind.EndOfFile;
+    }
+
+    take(): Token {
+        const result = { ...this };
+        this.scan();
+        return result;
+    }
+
+    *takeUntil(endToken: Kind, options?: { escape?: Kind[]; nestable?: [Kind, Kind][] }, yieldFinalClose?: boolean): Iterable<Token> {
+        const nestable = options?.nestable || [];
+        const escape = options?.escape || [];
+
+        processing: do {
+            switch (this.kind) {
+                case endToken:
+                    // if we're nested, we need to return the end token, because it's significant to the consumer
+                    if (yieldFinalClose) {
+                        yield this.take();
+                        return;
+                    }
+
+                    // we're done here, lose the last token and get out/

```

Comment:
Why does this comment end with a `/`? Can you remove that?

Reply:
removed